src/offline/cable_output.F90

```diff
@@ -3453,215 +2754,238 @@ SUBROUTINE create_restart(logn, dels, ktau, soil, veg, ssnow,
-!$                     ranges%swilt, .TRUE., 'real', .TRUE.)
+    !~ veg and soil
+    out_settings%dimswitch = "integer"
+    CALL check_and_write(output_var, rpid%iveg, &
+                         'iveg', REAL(veg%iveg, 4), ranges%iveg, patchout_var, out_settings)
+    CALL check_and_write(output_var, rpid%isoil, 'isoil', REAL(soil%isoilm, 4), &
+                         ranges%isoil, patchout_var, out_settings)
+    out_settings%dimswitch = "real"
+!$  CALL check_and_write(output_var, rpid%bch, 'bch', REAL(soil%bch, 4), &
+!$                       ranges%bch, patchout_var, out_settings)
+!$  CALL check_and_write(output_var, rpid%bch, 'bch', REAL(soil%bch, 4), &
+!$                       ranges%bch, patchout_var, out_settings)
+!$  CALL check_and_write(output_var, rpid%clay, 'clay', REAL(soil%clay, 4), &
+!$                       ranges%clay, patchout_var, out_settings)
+!$  CALL check_and_write(output_var, rpid%sand, 'sand', REAL(soil%sand, 4), &
+!$                       ranges%sand, patchout_var, out_settings)
+!$  CALL check_and_write(output_var, rpid%silt, 'silt', REAL(soil%silt, 4), &
+!$                       ranges%silt, patchout_var, out_settings)
+!$  CALL check_and_write(output_var, rpid%css, 'css', REAL(soil%css, 4), &
+!$                       ranges%css, patchout_var, out_settings)
+!$  CALL check_and_write(output_var, rpid%rhosoil, 'rhosoil', &
+!$                       REAL(soil%rhosoil, 4), ranges%rhosoil, patchout_var, &
+!$                       out_settings)
+!$  CALL check_and_write(output_var, rpid%hyds, 'hyds', REAL(soil%hyds, 4), &
+!$                       ranges%hyds, patchout_var, out_settings)
+!$  CALL check_and_write(output_var, rpid%sucs, 'sucs', REAL(soil%sucs, 4), &
+!$                       ranges%sucs, patchout_var, out_settings)
+!$  CALL check_and_write(output_var, rpid%rs20, 'rs20', REAL(veg%rs20, 4), &
+!$                       ranges%rs20, patchout_var, out_settings)
+!$  CALL check_and_write(output_var, rpid%ssat, 'ssat', REAL(soil%ssat, 4), &
+!$                       ranges%ssat, patchout_var, out_settings)
+!$  CALL check_and_write(output_var, rpid%sfc, 'sfc', REAL(soil%sfc, 4), &
+!$                       ranges%sfc, patchout_var, out_settings)
+!$  CALL check_and_write(output_var, rpid%swilt, 'swilt', REAL(soil%swilt, 4), &
+!$                       ranges%swilt, patchout_var, out_settings)
     ! Soil dimensioned variables/parameters:
-!$    CALL write_ovar (ncid_restart, rpid%froot, 'froot', REAL(veg%froot, 4),    &
-!$                     ranges%froot, .TRUE., 'soil', .TRUE.)
-    CALL write_ovar (ncid_restart, tggID, 'tgg', REAL(ssnow%tgg, 4),           &
-         ranges%SoilTemp, .TRUE., 'soil', .TRUE.)
-    CALL write_ovar (ncid_restart, wbID, 'wb', ssnow%wb, ranges%SoilMoist,     &
-         .TRUE., 'soil', .TRUE.)
-    CALL write_ovar (ncid_restart, wbiceID, 'wbice', ssnow%wbice,              &
-         ranges%SoilMoist, .TRUE., 'soil', .TRUE.)
```

Comment:
The variables `wb`, `wbice`, and `gammzz` are not being converted to `REAL(4)` so they have precision error in the output after the changes

Reply:
Yes, that's fine.